### PR TITLE
build(eslint-plugin-odyssey): allow numeric suffix

### PR DESCRIPTION
**Pull request by jeffbelser-okta**
### Description

This PR fine tunes our eslint rule to enforce valid theme token names. It also adjusts necessary lint escaping within `odyssey-react` to account for the rule changes

### CODE REVIEW

File: packages/odyssey-react/src/components/Text/Text.theme.ts [hunk 1]
```diff
@@ -26,12 +26,14 @@ export const theme: ThemeReducer = (theme) => ({
 
   LineHeight: theme.FontLineHeightBody,
   BodyLineHeight: theme.FontLineHeightBody,
+  /* eslint-disable @okta/odyssey/no-invalid-theme-properties */
   Heading1LineHeight: theme.FontLineHeightHeading1,
   Heading2LineHeight: theme.FontLineHeightHeading2,
   Heading3LineHeight: theme.FontLineHeightHeading3,
   Heading4LineHeight: theme.FontLineHeightHeading4,
   Heading5LineHeight: theme.FontLineHeightHeading5,
   Heading6LineHeight: theme.FontLineHeightHeading6,
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/Text/Text.theme.ts:35]: cc @edburyenegren-okta these tokens violate our numbers should only appear as suffix rule but not gonna worry too much for now

File: packages/odyssey-react/src/components/Heading/Heading.theme.ts [hunk 2]
```diff
@@ -27,6 +28,7 @@ export const theme: ThemeReducer = (theme) => ({
   Level4LineHeight: theme.FontLineHeightHeading4,
   Level5LineHeight: theme.FontLineHeightHeading5,
   Level6LineHeight: theme.FontLineHeightHeading6,
+  /* eslint-enable @okta/odyssey/no-invalid-theme-properties */
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/Heading/Heading.theme.ts:31]: cc @edburyenegren-okta these tokens violate our numbers should only appear as suffix rule but not gonna worry too much for now

File: packages/odyssey-react/src/components/Text/Text.theme.ts (hunk 1 above)
Comment (edburyenegren-okta) [reply packages/odyssey-react/src/components/Text/Text.theme.ts:35]: I thought we were changing the rule specifically to allow mid-token numbers so the modifier matches the token format.

Ex: Line 31 is the `line-height` property for `Heading3`, not the third variation of `Heading` `line-height` (as in `HeadingLineHeight3`).

Comment (edburyenegren-okta) [reply packages/odyssey-react/src/components/Text/Text.theme.ts:35]: Ugh, maybe our wires got crossed at some point. At the time, I thought these were still escaped/disallowed, which is why we had `Remove any unnecessary rule escapes` as one of the AC. 😓

**Review by edburyenegren-okta — APPROVED** [review]
👏🏼

**Pull request merged**